Added FunctionCall support

MFAshby opened the pull request:
Added support for FunctionCall message as per
https://www.postgresql.org/docs/11/protocol-message-formats.html

Adds unit test for Encode / Decode cycle and invalid message format
errors.

Fixes https://github.com/jackc/pgproto3/issues/23

jackc commented:
I'm getting a test failure:

```
jack@glados ~/dev/pgproto3 ±patientsknowbest-add_function_call » go test                                                                                                                                                                
--- FAIL: TestFunctionCall_EncodeDecode (0.00s)
    --- FAIL: TestFunctionCall_EncodeDecode/invalid_format_code (0.00s)
        function_call_test.go:40: [ArgFormatCodes: [2 1 0 1] != <nil slice> Arguments: [[102 111 111] [98 97 114] [98 97 122]] != <nil slice>]
    --- FAIL: TestFunctionCall_EncodeDecode/invalid_result_format_code (0.00s)
        function_call_test.go:40: [ResultFormatCode: 2 != 0]
FAIL
exit status 1
FAIL	github.com/jackc/pgproto3/v2	0.114s
```

Also, would it be possible to use testify or reflect.DeepEqual instead of adding `github.com/go-test/deep`? I'm (belated) trying to tame the pgx family dependency tree.

MFAshby commented:
@jackc thanks for the feedback. I totally forgot to run 'go test' after adding more test cases. 

I've fixed it, removed the dependency as you suggested, and extended the test a little to check the message header.

gitstashpop commented:
@MFAshby can you please run [gofmt](https://go.dev/blog/gofmt) on your code? You can set up your editor to do this on save for future Go development.
